apps/web/src/utils/segment.ts

```diff
@@ -1,14 +1,26 @@
 import { AnalyticsBrowser } from '@segment/analytics-next';
 import { IUserEntity } from '@novu/shared';
 import * as mixpanel from 'mixpanel-browser';
-import { api } from '../api';
+import { UseMutateFunction } from '@tanstack/react-query';
+
+type TrackFunction = UseMutateFunction<
+  any,
+  unknown,
+  {
+    event: string;
+    data?: Record<string, unknown> | undefined;
+  },
+  unknown
+>;
 
 export class SegmentService {
   private _segment: AnalyticsBrowser | null = null;
+  private readonly _trackFunction: TrackFunction;
   private _segmentEnabled: boolean;
   public _mixpanelEnabled: boolean;
 
-  constructor() {
+  constructor(trackFunction: TrackFunction) {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** Convert this to an object to make it easier to extend with a smaller changeset in the future‏